src/generators/Generator.js

```diff
@@ -260,14 +260,26 @@ export default class Generator {
 					// export is last property, we can just return it
 					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `return ` );
 				} else {
-					// TODO ensure `template` isn't already declared
-					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `var template = ` );
+					// we need to avoid a conflict with anything on the top-level scope of the component
+					// however, determining which things these are is tricky, so we instead just avoid conflicts with any identifiers anywhere
```

Comment:
Ah, yeah — it should skip `ExportDefaultDeclaration` nodes. [This regex](https://github.com/sveltejs/svelte/blob/69ee3e613905eb0a700699bd13884603d0b2d3cb/src/generators/annotateWithScopes.js#L31) should probably be `/(Function|Class|Variable)Declaration/` instead